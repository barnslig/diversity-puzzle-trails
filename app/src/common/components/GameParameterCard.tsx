import { Box, Card, CardContent, List, Typography } from "@mui/material";
import * as React from "react";

type GameParameterCardProps = {
  /**
   * Common title of the presented game parameters
   */
  title: React.ReactNode;

  /**
   * Common description of the game parameters
   */
  description: React.ReactNode;

  /**
   * Put <GameParameter />s in here :)
   */
  children: React.ReactNode;
};

/**
 * A card presenting multiple game parameters
 */
const GameParameterCard = ({
  title,
  description,
  children,
}: GameParameterCardProps) => {
  return (
    <Card component="article" sx={{ marginBottom: 2 }}>
      <CardContent>
        <Box component="header" sx={{ marginBottom: 1 }}>
          <Typography sx={{ marginBottom: 1 }} variant="h5" component="h2">
            {title}
          </Typography>
          <Typography variant="body2" color="textSecondary">
            {description}
          </Typography>
        </Box>
        <List disablePadding>{children}</List>
      </CardContent>
    </Card>
  );
};

export default GameParameterCard;
