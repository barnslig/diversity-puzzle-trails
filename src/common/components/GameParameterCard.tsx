import {
  Card,
  CardContent,
  List,
  makeStyles,
  Typography,
} from "@material-ui/core";
import * as React from "react";

const useStyle = makeStyles((theme) => ({
  card: {
    marginBottom: theme.spacing(2),
  },
  cardHeader: {
    marginBottom: theme.spacing(1),
  },
  cardHeaderHl: {
    marginBottom: theme.spacing(1),
  },
}));

type GameParameterCardProps = {
  /**
   * Common title of the presented game parameters
   */
  title: React.ReactNode;

  /**
   * Common description of the game parameters
   */
  description: React.ReactNode;

  /**
   * Put <GameParameter />s in here :)
   */
  children: React.ReactNode;
};

/**
 * A card presenting multiple game parameters
 */
const GameParameterCard = ({
  title,
  description,
  children,
}: GameParameterCardProps) => {
  const classes = useStyle();

  return (
    <Card className={classes.card} component="article">
      <CardContent>
        <header className={classes.cardHeader}>
          <Typography
            className={classes.cardHeaderHl}
            variant="h5"
            component="h2"
          >
            {title}
          </Typography>
          <Typography variant="body2" color="textSecondary">
            {description}
          </Typography>
        </header>
        <List disablePadding>{children}</List>
      </CardContent>
    </Card>
  );
};

export default GameParameterCard;
