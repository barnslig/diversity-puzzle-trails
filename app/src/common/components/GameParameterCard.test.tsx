import { render } from "@testing-library/react";
import * as React from "react";

import GameParameterCard from "./GameParameterCard";

it("renders correctly", () => {
  const { container } = render(
    <GameParameterCard
      title="Spielstand"
      description="Der Status eurer gemeinsamen Mission. Achtet darauf, dass kein Wert 0 erreicht!"
    >
      <div>hypothetical GameParameter</div>
      <div>hypothetical GameParameter</div>
    </GameParameterCard>
  );
  expect(container.firstChild).toMatchSnapshot();
});
