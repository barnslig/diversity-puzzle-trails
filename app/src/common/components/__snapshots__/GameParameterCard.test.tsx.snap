// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<article
  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-xaep3o-MuiPaper-root-MuiCard-root"
>
  <div
    class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <header
      class="MuiBox-root css-19idom"
    >
      <h2
        class="MuiTypography-root MuiTypography-h5 css-1jcm1rz-MuiTypography-root"
      >
        Spielstand
      </h2>
      <p
        class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
      >
        Der Status eurer gemeinsamen Mission. Achtet darauf, dass kein Wert 0 erreicht!
      </p>
    </header>
    <ul
      class="MuiList-root css-1mk9mw3-MuiList-root"
    >
      <div>
        hypothetical GameParameter
      </div>
      <div>
        hypothetical GameParameter
      </div>
    </ul>
  </div>
</article>
`;
